classes/class-woothemes-sensei-certificates.php

```diff
@@ -1483,24 +1483,36 @@ public function enqueue_block_editor_assets() {
 	 * @return string Block HTML.
 	 */
 	public function update_view_certificate_button_url( $block_content, $block ): string {
+		$class_name = 'view-certificate';
+
 		// phpcs:ignore WordPress.Security.NonceVerification.Recommended -- Only used if the learner completed the course.
 		$course_id = isset( $_GET['course_id'] ) ? (int) $_GET['course_id'] : false;
 
-		// Check that the course ID exists and that the user has completed the course.
 		if (
+			// Check that the course ID exists and that the user has completed the course.
 			! $course_id
 			|| ! get_current_user_id()
 			|| 'course' !== get_post_type( $course_id )
 			|| ! Sensei_Utils::user_completed_course( $course_id, get_current_user_id() )
+
+			// Check that the block is a core/button and it contains the respective class name.
+			|| ! isset( $block['blockName'] )
+			|| 'core/button' !== $block['blockName']
+			|| ! isset( $block['attrs']['className'] )
+			|| false === strpos( $block['attrs']['className'], $class_name )
```

Comment:
Technically, this is not an exact match. For example, a button with the class `should-view-certificates` is matched here, and follows the same logic. This also applies to the identical logic in `update_button_block_url`.

An exact match using a regex would be tighter. I'm not sure it's super important though 🤷‍♂️

Reply:
That's true! We're using it in many different cases of the code (even in the Sensei core). I think we could keep it as it is for now, and refactor all the cases to use regex if we have any user with problems. We could create a single helper function in the core for that instead replicate the same regex in many different places.
WDYT? If you think it's important, I can do it, but I'd do it in a different PR to handle all the cases.